code/game/machinery/cryopod.dm

```diff
@@ -0,0 +1,436 @@
+			O.team.objectives -= O
+			qdel(O)
+			for(var/datum/mind/M in O.team.members)
+				to_chat(M.current, "<BR><span class='userdanger'>Your target is no longer within reach. Objective removed!</span>")
+				M.announce_objectives()
+		else if(O.target && istype(O.target, /datum/mind))
+			if(O.target == mob_occupant.mind)
+				var/old_target = O.target
+				O.target = null
+				if(!O)
+					return
+				O.find_target()
+				if(!O.target && O.owner)
+					to_chat(O.owner.current, "<BR><span class='userdanger'>Your target is no longer within reach. Objective removed!</span>")
+					for(var/datum/antagonist/A in O.owner.antag_datums)
+						A.objectives -= O
+				if (!O.team)
+					O.update_explanation_text()
+					O.owner.announce_objectives()
+					to_chat(O.owner.current, "<BR><span class='userdanger'>You get the feeling your target is no longer within reach. Time for Plan [pick("A","B","C","D","X","Y","Z")]. Objectives updated!</span>")
+				else
+					var/list/objectivestoupdate
+					for(var/datum/mind/own in O.get_owners())
+						to_chat(own.current, "<BR><span class='userdanger'>You get the feeling your target is no longer within reach. Time for Plan [pick("A","B","C","D","X","Y","Z")]. Objectives updated!</span>")
+						for(var/datum/objective/ob in own.get_all_objectives())
+							LAZYADD(objectivestoupdate, ob)
+					objectivestoupdate += O.team.objectives
+					for(var/datum/objective/ob in objectivestoupdate)
+						if(ob.target != old_target || !istype(ob,O.type))
+							return
+						ob.target = O.target
+						ob.update_explanation_text()
+						to_chat(O.owner.current, "<BR><span class='userdanger'>You get the feeling your target is no longer within reach. Time for Plan [pick("A","B","C","D","X","Y","Z")]. Objectives updated!</span>")
+						ob.owner.announce_objectives()
+				qdel(O)
+
+// This function can not be undone; do not call this unless you are sure
+/obj/machinery/cryopod/proc/despawn_occupant()
+	var/mob/living/mob_occupant = occupant
+
+	if(mob_occupant.mind && mob_occupant.mind.assigned_role)
+		//Handle job slot/tater cleanup.
+		var/job = mob_occupant.mind.assigned_role
+		SSjob.FreeRole(job)
+		if(LAZYLEN(mob_occupant.mind.objectives))
+			mob_occupant.mind.objectives.Cut()
+			mob_occupant.mind.special_role = null
+
+	// Delete them from datacore.
+
+	var/announce_rank = null
+	for(var/datum/data/record/R in GLOB.data_core.medical)
+		if((R.fields["name"] == mob_occupant.real_name))
+			qdel(R)
+	for(var/datum/data/record/T in GLOB.data_core.security)
+		if((T.fields["name"] == mob_occupant.real_name))
+			qdel(T)
+	for(var/datum/data/record/G in GLOB.data_core.general)
+		if((G.fields["name"] == mob_occupant.real_name))
+			announce_rank = G.fields["rank"]
+			qdel(G)
+
+	for(var/obj/machinery/computer/cloning/cloner in world)
+		for(var/datum/data/record/R in cloner.records)
+			if(R.fields["name"] == mob_occupant.real_name)
+				cloner.records.Remove(R)
```

Comment:
Remove this shit, just make the ghost unable to reenter body and cloning will fail